jdat_notebooks/MRS_Mstar_analysis/JWST_Mstar_dataAnalysis_PointSourceDetectorBasedExtraction.ipynb

```diff
@@ -0,0 +1,2258 @@
+{
```

Comment:
<p>Again, using lists and a loop could make this cell and others like it easier to read and maintain.</p><p> </p><p>If you want to keep a similar, <code>urllib</code>-based structure, you could gather the intended file names and URLS into lists and loop through them like so:</p><p><code></code><code></p><p>from pathlib import Path # would move to top with other imports</p><p> </p><p>data_dir = Path('./miri_devel/')</p><p>data_paths = ['folder11/folder12/file1',</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;'folder21/folder22/file2'] # and so on</p><p>data_paths_full = [data_dir / p for p in data_paths]</p><p>data_urls = ['https://data.sceince.stsci.edu/file1',</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;'https://data.sceince.stsci.edu/file2'] # and so on</p><p> </p><p>for p, url in zip(data_paths_full, data_urls):</p><p>&nbsp;&nbsp;&nbsp;if os.path.exists(p):</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;print(f"{p.name} exists.")</p><p>&nbsp;&nbsp;&nbsp;else:</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;urllib.request.urlretrieve(url, p)</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;print(f"Downloading {p.name}...")</p><p></code><code></code></p><p> </p><p>The above code would only need minor tweaks to work for neighboring cells, too.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/dat_pyinthesky/pull/162/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Reply:
```
from pathlib import Path # would move to top with other imports

data_dir = Path('./miri_devel/')
data_paths = ['folder11/folder12/file1',
              'folder21/folder22/file2'] # and so on
data_paths_full = [data_dir / p for p in data_paths]
data_urls = ['https://data.sceince.stsci.edu/file1',
             'https://data.sceince.stsci.edu/file2'] # and so on

for p, url in zip(data_paths_full, data_urls):
    if os.path.exists(p):
        print(f"{p.name} exists.")
    else:
        urllib.request.urlretrieve(url, p)
        print(f"Downloading {p.name}...")
```

The above code would only need minor tweaks to work for neighboring cells, too.

Reply:
Done